### PR TITLE
Fix mypy errors in postgres/hooks and postgres/operators

### PR DESCRIPTION
**Pull request by subkanthi**
related: #19891 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/postgres/hooks/postgres.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def get_iam_token(self, conn: Connection) -> Tuple[str, str, int]:
             # Pull the custer-identifier from the beginning of the Redshift URL
             # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
             cluster_identifier = conn.extra_dejson.get('cluster-identifier', conn.host.split('.')[0])
-            session, endpoint_url = aws_hook._get_credentials()
+            session, endpoint_url = aws_hook._get_credentials(region_name=None)
```
Comment (subkanthi) [inline airflow/providers/postgres/hooks/postgres.py:195]: was just surprised to see aws_hook in postgres, Im sure its done for a reason.

Comment (potiuk) [reply airflow/providers/postgres/hooks/postgres.py:195]: Yep. This is to support RedShift.

**Review by potiuk — APPROVED** [review]

File: airflow/providers/postgres/hooks/postgres.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/postgres/hooks/postgres.py:195]: Wait. I think this might be not backwards compatible....

Comment (potiuk) [reply airflow/providers/postgres/hooks/postgres.py:195]: No. It's OK.

**Review by potiuk — APPROVED** [review]

**Pull request merged**